cmd/hotfixreleaser.go

```diff
@@ -71,11 +94,16 @@ func (h *hotfixer) runHotfixer(cmd *cobra.Command, _ []string) error {
 			continue
 		}
 
+		// let's skip if the the serviceID is not empty and matches the service
+		if serviceID != "" && xrd.GetLabels()[serviceIDLabel] != serviceID {
+			continue
+		}
+
 		p := fieldpath.Pave(xrd.Object)
 
 		XRDLabels := xrd.GetLabels()
 
-		if XRDLabels[release.ServiceIDLabel] == "" {
+		if XRDLabels[serviceIDLabel] == "" {
```

Comment:
Should we really abort here when there is one not labeled?

Reply:
I'd argue for that yes. It's something we have to wire-up via the component when we generate the XRDs. So if we deploy a release that has a broken XRD label, something might be very not okay.